front/app/containers/MainHeader/Components/DesktopNavItems/index.tsx

```diff
@@ -38,18 +44,267 @@ const NavbarItems = styled.ul`
+          visibleWithMore.push(allItems[index]);
+          widthWithMore += itemWidth;
+        } else {
+          overflowWithMore.push(allItems[index]);
+        }
+      });
+
+      // Ensure "All projects" is always in visible items
+      if (allProjectsIndex !== -1) {
+        const allProjectsInOverflow = overflowWithMore.some(
+          (item) => item.linkTo === '/projects'
+        );
+
+        if (allProjectsInOverflow) {
+          // Recalculate visible items to make room for "All projects"
+          const recalculatedVisible: NavbarItemProps[] = [];
+          const recalculatedOverflow: NavbarItemProps[] = [];
+          let recalculatedWidth = 0;
+
+          tempItems.forEach((tempItem, index) => {
+            const itemWidth = tempItem.offsetWidth;
+            const isAllProjects = allItems[index].linkTo === '/projects';
+
+            if (isAllProjects) {
+              // Always include "All projects" in visible
+              recalculatedVisible.push(allItems[index]);
+              recalculatedWidth += itemWidth;
+            } else {
+              const wouldFit =
+                recalculatedWidth + itemWidth <=
+                availableWidth - moreButtonWidth;
+              if (wouldFit) {
+                recalculatedVisible.push(allItems[index]);
+                recalculatedWidth += itemWidth;
+              } else {
+                recalculatedOverflow.push(allItems[index]);
+              }
+            }
+          });
+
+          if (
+            !arraysEqual(visibleItems, recalculatedVisible) ||
+            !arraysEqual(overflowItems, recalculatedOverflow)
+          ) {
+            setVisibleItems(recalculatedVisible);
+            setOverflowItems(recalculatedOverflow);
+          }
+        } else {
+          if (
+            !arraysEqual(visibleItems, visibleWithMore) ||
+            !arraysEqual(overflowItems, overflowWithMore)
+          ) {
+            setVisibleItems(visibleWithMore);
+            setOverflowItems(overflowWithMore);
+          }
+        }
+      } else {
+        if (
+          !arraysEqual(visibleItems, visibleWithMore) ||
+          !arraysEqual(overflowItems, overflowWithMore)
+        ) {
+          setVisibleItems(visibleWithMore);
+          setOverflowItems(overflowWithMore);
+        }
+      }
+    }
+
+    tempItems.forEach((item) => item.remove());
+  }, [navbarItems, visibleItems, overflowItems]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      calculateVisibleItems();
+    }, 100);
+
+    let resizeTimer: NodeJS.Timeout;
+    const handleResize = () => {
+      clearTimeout(resizeTimer);
+      resizeTimer = setTimeout(() => {
+        calculateVisibleItems();
+      }, 250); // Increased debounce time
+    };
+
+    // Listen for dropdown state changes
+    document.addEventListener('click', (e) => {
+      const target = e.target as Element;
+      if (target.closest('[data-testid="admin-publications-navbar-item"]')) {
```

Comment:
I don't think we should rely on `data-testid` properties for anything that is not testing. The property suggesets that it is only relevant for tests. So if you would delete the test, you would assume that you can also safely delete this property without breaking anything.